src/eureka/S1_detector_processing/bias_sub.py

```diff
@@ -0,0 +1,187 @@
+            data_median = np.nanmedian(data[jj][np.where(trace_mask)])
+            scale_factor[ii, jj] = data_median/bias_median
+    mean_scale_factor = np.mean(scale_factor, axis=0)
+    log.writelog(f'  Mean bias scale factors (by group): {mean_scale_factor}')
+    
+    # Get segment number
+    segment = str(input_model.meta.exposure.segment_number).zfill(3)
+
+    # Save scale factor to ECSV file
+    fname = meta.outputdir+"S1_seg"+segment+"_BiasScaleFactor.ecsv"
+    savetable_S1(fname, scale_factor)
+
+    # Smooth scale factor
+    if meta.bias_correction == "smooth":
+        for jj in range(ngroup):
+            scale_factor[:, jj] = smooth(scale_factor[:, jj], 
+                                         meta.bias_smooth_length)
+        # Save smoothed scale factor to ECSV file
+        fname = meta.outputdir+"S1_seg"+segment+"_BiasSmoothedScaleFactor.ecsv"
+        savetable_S1(fname, scale_factor)
+
+    # Replace NaN's in the superbias with zeros
+    bias_model.data[np.isnan(bias_model.data)] = 0.0
+
+    # Create 4D bias array for each integration/group and apply scale factor 
+    bias_data = np.ones((nint, ngroup, nrow, ncol)) * \
+        bias_model.data[np.newaxis, np.newaxis, :, :]
+    
+    try:
+        if meta.bias_correction == "mean":
+            if isinstance(meta.bias_group, int):
+                # Scale superbias frame using single value for all groups
+                jj = meta.bias_group
+                assert jj > 0, f"Group number should be >0, got: {jj}"
+                assert jj <= ngroup, f"Group number should be <={ngroup}" + \
+                    ", got: {jj}"
+                log.writelog('  Applying mean bias correction using ' +
+                             f'group {jj}.')
+                bias_model.data *= mean_scale_factor[jj-1]
+            elif meta.bias_group == "each":
+                # Scale superbias frame using means values for each group
+                log.writelog('  Applying mean bias correction to each group.')
+                bias_model.data *= mean_scale_factor[0]
+                bias_data *= mean_scale_factor[np.newaxis, :, np.newaxis, 
+                                               np.newaxis]
+            else:
+                raise ValueError('Incorrect meta.bias_group value: ' + 
+                                 f'{meta.bias_group}. Should be ' +
+                                 '[1, 2, ..., each].')
+        elif (meta.bias_correction == "group_level") or \
+             (meta.bias_correction == "smooth"):
+            # Apply correction for zeroframe
+            bias_model.data *= mean_scale_factor[0]
+            if isinstance(meta.bias_group, int):
+                # Scale superbias frame using values from one group
+                jj = meta.bias_group
+                assert jj > 0, f"Group number should be >0, got: {jj}"
+                assert jj <= ngroup, f"Group number should be <={ngroup}" + \
+                    ", got: {jj}"
```

Comment:
I'm pretty sure this needs to be an fstring as well as I think this would just print "{jj}" instead of the value of jj